core/chains/evm/txmgr/transmitchecker.go

```diff
@@ -134,6 +139,13 @@ type VRFV1Checker struct {
 	// Callbacks checks whether a VRF V1 request has already been fulfilled on the VRFCoordinator
 	// Solidity contract
 	Callbacks func(opts *bind.CallOpts, reqID [32]byte) (v1.Callbacks, error)
+
+	// HeaderByNumber fetches the header given the number. If nil is provided,
+	// the latest header is fetched.
+	HeaderByNumber func(ctx context.Context, n *big.Int) (*gethtypes.Header, error)
```

Comment:
I think we should use `HeadByNumber` here since it's technically more correct - `HeaderByNumber` will return the wrong blockhash for e.g avalanche.